src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -390,54 +390,23 @@ private static bool IsObjectPropertyNameContext(List<SyntaxBase> matchingNodes,
             return false;
         }
 
-        private static bool IsPropertyValueContext(List<SyntaxBase> matchingNodes, (ObjectPropertySyntax? node, int index) propertyInfo)
-        {
-            // find the innermost property
-            if (propertyInfo.node == null)
-            {
-                // none of the nodes are object properties,
-                // so we can't possibly be in a property value context
-                return false;
-            }
-
-            switch (matchingNodes[^1])
-            {
-                case ObjectPropertySyntax _:
-                    // the cursor position may be in the trivia following the colon that follows the property name
-                    // if that's the case, the offset should match the end of the property span exactly
-                    return true;
-
-                case Token token:
-                    // how many matching nodes remain including the object node itself
-                    int nodeCount = matchingNodes.Count - propertyInfo.index;
-
-                    switch (nodeCount)
-                    {
-                        case 2 when token.Type == TokenType.Colon:
-                        {
-                            // the cursor position is after the colon that follows the property name
-                            return true;
-                        }
+        private static bool IsPropertyValueContext(List<SyntaxBase> matchingNodes, (ObjectPropertySyntax? node, int index) propertyInfo) =>
+            propertyInfo.node != null &&
+                (
+                // the cursor position may be in the trivia following the colon that follows the property name
+                // if that's the case, the offset should match the end of the property span exactly
+                SyntaxMatcher.IsTailMatch<ObjectPropertySyntax>(matchingNodes, _ => true) ||
 
-                        case 3 when matchingNodes[^2] is StringSyntax stringSyntax && ReferenceEquals(propertyInfo.node.Value, stringSyntax):
-                        {
-                            // the cursor is inside a string value of the property
-                            return true;
-                        }
+                // the cursor position is after the colon that follows the property name
+                SyntaxMatcher.IsTailMatch<ObjectPropertySyntax, Token>(matchingNodes, (_, token) => token.Type == TokenType.Colon) ||
 
-                        case 4 when matchingNodes[^2] is IdentifierSyntax identifier && ReferenceEquals(propertyInfo.node.Value, identifier):
```

Comment:
The minimal fix would have been to change L428 to 
`case 4 when matchingNodes[^3] is VariableAccessSyntax variableAccess && matchingNodes[^2] is IdentifierSyntax _ && ReferenceEquals(propertyInfo.node.Value, variableAccess):
 
However, I've been opportunistically converting to the tail matcher pattern since it's more readable once you get used to it.